### PR TITLE
Post Controller 

### PR DESCRIPTION
**Pull request by janko101**
### PostController
- Add RequestMapping for class
- PostMapping for creating a new post
- GetMapping for returning all posts
- GetMapping for returning a post by ID
- PutMapping for updating a post by ID
- DeleteMapping for deleting a post by ID

### User Story
```
As a user of Cuddly 
I want to create a post with texts
so that I can share contents about my pet

### CODE REVIEW

File: src/main/java/se/kth/sda/skeleton/auth/AuthController.java [hunk 1]
```diff
@@ -37,4 +37,5 @@
             return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
         }
     }
+
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/auth/AuthController.java]: Perhaps you could revert this new line so all the files changed are only related to Post.

File: src/main/java/se/kth/sda/skeleton/posts/Post.java [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
+@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
 public class Post {
 
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
 
+    @NotBlank
+    @Column(nullable = false)
     private String body;
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/posts/Post.java]: IGNORE In the JSON fields sheet, we have the property name **contentText** for the post content, but I think **body** is fine to keep and we update the JSON doc.

Comment (NeuralAlchemist) [inline src/main/java/se/kth/sda/skeleton/posts/Post.java]: Hey! Would you change this to "contentText" so the fields are the same as the names we've decided in the JSON Fields sheet?

File: src/main/java/se/kth/sda/skeleton/posts/Post.java [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
 
+    // Constructor
     public Post() {
     }
 
+    // Getters and Setters
     public Post(String body) {
         this.body = body;
```
Comment (NeuralAlchemist) [inline src/main/java/se/kth/sda/skeleton/posts/Post.java]: Hey! Would you change body to "contentText" so the fields are the same as the names we've decided in the JSON Fields sheet?

File: src/main/java/se/kth/sda/skeleton/posts/PostController.java [hunk 4]
```diff
@@ -1,7 +1,79 @@
...
+
+import javax.validation.Valid;
+import java.util.List;
 
 @RestController
+@RequestMapping("/posts")
 public class PostController {
+    PostService postService;
+
+    @Autowired
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/posts/PostController.java:25]: Suggested javadoc:

/**
     * Constructs a PostController and automatically assigns its postService field.
     * 
     * @param postService an object that implements interface PostService
     */

File: src/main/java/se/kth/sda/skeleton/posts/Post.java (hunk 2 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/posts/Post.java]: Changed

File: src/main/java/se/kth/sda/skeleton/posts/Post.java (hunk 3 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/posts/Post.java]: Changed

File: src/main/java/se/kth/sda/skeleton/posts/PostService.java [hunk 5]
```diff
@@ -1,7 +1,47 @@
 package se.kth.sda.skeleton.posts;
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/posts/PostService.java:1]: At some point, we should probably update the folder names. Perhaps something for the backlog.

**Review by eeels22 — CHANGES_REQUESTED** [review]
Thanks Janko. Code is looking good. Just wonder about completing javadocs for all classes and non-trivial methods to meet the definition of done.

File: src/main/java/se/kth/sda/skeleton/posts/Post.java (hunk 2 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/posts/Post.java]: Should I add some anotation here?

Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/posts/Post.java]: that jsonIgnore thingy?

File: src/main/java/se/kth/sda/skeleton/posts/PostService.java [hunk 6]
```diff
@@ -1,7 +1,76 @@
...
 
 @Service
 public class PostService {
+
+    PostRepository postRepository;
+
+    @Autowired
+    public PostService(PostRepository postRepository) {
+        this.postRepository = postRepository;
+    }
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/posts/PostService.java:26]: Suggested javadoc:

/**
     * Constructs a PostService and automatically assigns its postRepository field
     * 
     * @param postRepository an object that implements interface PostRepository
     */

File: src/main/java/se/kth/sda/skeleton/posts/PostController.java (hunk 4 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/posts/PostController.java:25]: added

File: src/main/java/se/kth/sda/skeleton/posts/PostService.java (hunk 6 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/posts/PostService.java:26]: added

File: src/main/java/se/kth/sda/skeleton/posts/PostService.java (hunk 5 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/posts/PostService.java:1]: yap...was looking at that

File: src/main/java/se/kth/sda/skeleton/auth/AuthController.java (hunk 1 above)
Comment (janko101) [reply src/main/java/se/kth/sda/skeleton/auth/AuthController.java]: good catch...Done

**Review by NeuralAlchemist — APPROVED** [review]
Looks good thanks for adding the docs

**Review by eeels22 — APPROVED** [review]
Thanks Janko!

**Pull request merged**